rasa/shared/core/generator.py

```diff
@@ -217,6 +217,40 @@ def update(self, event: Event, skip_states: bool = False) -> None:
 
             self._append_current_state()
 
+    def fingerprint(self) -> Text:
+        """Returns a unique hash for the tracker which is stable across python runs.
+
+        Returns:
+            fingerprint of the tracker
+        """
+        data = []
+        sender_id_hash = rasa.shared.utils.io.get_text_hash(self.sender_id)
+        data.append(sender_id_hash)
+
+        if self.domain:
+            data.append(self.domain.fingerprint())
+
+        if self.slots:
+            for slot in self.slots.values():
+                data.append(slot.fingerprint())
```

Comment:
Maybe `data` should be a `dict` rather than a list?. We probably shouldn't rely on `slots.values()` always giving the same order, and we don't want that to change the fingerprint.